apis/v1beta1/config.go

```diff
@@ -465,9 +465,22 @@ func (s *Service) MetricsEndpoint(logger logr.Logger) (string, int32, error) {
 
 // ApplyDefaults inserts configuration defaults if it has not been set.
 func (s *Service) ApplyDefaults(logger logr.Logger) error {
-	tel := s.GetTelemetry()
+	if s.Telemetry != nil {
+		logger.V(2).Info("original telemetry configuration before parsing", "rawConfig", s.Telemetry.Object)
+
+		if metricsConfig, ok := s.Telemetry.Object["metrics"]; ok {
+			if metricsMap, ok := metricsConfig.(map[string]interface{}); ok {
+				if readers, hasReaders := metricsMap["readers"]; hasReaders && readers != nil {
+					logger.Info("found telemetry readers in original configuration", "readers", readers)
+				}
+			}
+		}
+	}
```

Comment:
Why are we doing the manual check here given we do the same thing below?

Reply:
Just removed in favor of the logs below.